### PR TITLE
[JP install full plugin] Implement onboarding tracking

### LINKED ISSUE #17833: Create the fullscreen modal
Refs qk5s4q4mZsXknwhMWo2vIY-fi-8221%3A20788

Image

## Tasks

- [x] Create the fullscreen modal UI.
- [ ] Show the Jetpack plugin install flow when the Continue button is tapped.
- [x] Dismiss the modal and show My Sites when the Close button is tapped.
- [x] Show the Help screen when the Contact Support button is tapped.
- [x] Implement the presentation logic.

### PR DESCRIPTION
**Pull request by RenanLukas**
Fixes #17833 

Should be merged after https://github.com/wordpress-mobile/WordPress-Android/pull/17949

To test:
This screen has no entry point yet. It will be testable when install full plugin "My Site" card is implemented.
To test the screen in this PR, you can use the following patch to add a test entry point:

```
diff --git a/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java b/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
index 5903791640..23118c0b3f 100644
--- a/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
@@ -91,6 +91,7 @@ import org.wordpress.android.ui.jetpackoverlay.JetpackFeatureFullScreenOverlayFr
 import org.wordpress.android.ui.jetpackoverlay.JetpackFeatureRemovalOverlayUtil;
 import org.wordpress.android.ui.jetpackoverlay.JetpackFeatureRemovalOverlayUtil.JetpackFeatureCollectionOverlaySource;
 import org.wordpress.android.ui.jetpackoverlay.JetpackFeatureRemovalPhaseHelper;
+import org.wordpress.android.ui.jpfullplugininstall.onboarding.JetpackFullPluginInstallOnboardingDialogFragment;
 import org.wordpress.android.ui.main.WPMainNavigationView.OnPageListener;
 import org.wordpress.android.ui.main.WPMainNavigationView.PageType;
 import org.wordpress.android.ui.mlp.ModalLayoutPickerFragment;
@@ -541,8 +542,10 @@ public class WPMainActivity extends LocaleAwareActivity implements
     }

     private void showBloggingPromptsOnboarding() {
-        BloggingPromptsOnboardingDialogFragment.newInstance(DialogType.ONBOARDING).show(
-                getSupportFragmentManager(), BloggingPromptsOnboardingDialogFragment.TAG
+//        BloggingPromptsOnboardingDialogFragment.newInstance(DialogType.ONBOARDING).show(
+//                getSupportFragmentManager(), BloggingPromptsOnboardingDialogFragment.TAG));
+        JetpackFullPluginInstallOnboardingDialogFragment.newInstance().show(
+                getSupportFragmentManager(), JetpackFullPluginInstallOnboardingDialogFragment.TAG
         );
     }
```
With the patch applied:
1 - Install Jetpack app;
2 - Select a site that is a blog;
3 - Tap the overflow menu in the prompt card;
4 - Select `"Learn more"`;
5 - When the screen is loaded, the analytics event `jp_install_full_plugin_onboarding_modal_viewed` should be tracked;
6 - When the install button is tapped, the analytics event `jp_install_full_plugin_onboarding_modal_install_tapped` should be tracked;
7 - When the screen is dismissed by tapping on the "X" button, the analytics event `jp_install_full_plugin_onboarding_modal_dismissed` should be tracked;
8 - When the screen is dismissed by tapping the back button, the analytics event `jp_install_full_plugin_onboarding_modal_dismissed` should be tracked.

```
com.jetpack.android.prealpha I/WordPress-STATS: 🔵 Tracked: jp_install_full_plugin_onboarding_modal_viewed
com.jetpack.android.prealpha I/WordPress-STATS: 🔵 Tracked: jp_install_full_plugin_onboarding_modal_install_tapped
com.jetpack.android.prealpha I/WordPress-STATS: 🔵 Tracked: jp_install_full_plugin_onboarding_modal_dismissed
```

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit testing

3. What automated tests I added (or what prevented me from doing so)
`JetpackFullPluginInstallOnboardingAnalyticsTrackerTest.kt` and `JetpackFullPluginInstallOnboardingViewModelTest.kt`.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/jpfullplugininstall/onboarding/JetpackFullPluginInstallOnboardingViewModelTest.kt [hunk 1]
```diff
@@ -103,6 +105,18 @@ class JetpackFullPluginInstallOnboardingViewModelTest : BaseUnitTest() {
...
+        classToTest.onInstallFullPluginClick()
+        verify(analyticsTracker).trackInstallButtonClick()
+    }
+
+    @Test
+    fun `Should track dismiss screen click when onDismissScreenClick is called`() {
+        classToTest.onDismissScreenClick()
+        verify(analyticsTracker).trackScreenDismissed()
+    }
+
```
Comment (develric) [inline WordPress/src/test/java/org/wordpress/android/ui/jpfullplugininstall/onboarding/JetpackFullPluginInstallOnboardingViewModelTest.kt:119]: Should we add a unit test for the trackScreenShown as well? (and as a side note, thanks for always and consistently adding unit tests 👍 !)

**Review by develric — COMMENTED** [review]
Thanks for the PR @RenanLukas 👍 ! I just left a comment about considering adding an additional unit test; let me know wdyt and I'm happy to move this forward 🙇

File: WordPress/src/test/java/org/wordpress/android/ui/jpfullplugininstall/onboarding/JetpackFullPluginInstallOnboardingViewModelTest.kt (hunk 1 above)
Comment (RenanLukas) [reply WordPress/src/test/java/org/wordpress/android/ui/jpfullplugininstall/onboarding/JetpackFullPluginInstallOnboardingViewModelTest.kt:119]: Thanks for catching that! I missed that one 🤦 (thanks 🙇 )

**Review by develric — APPROVED** [review]
Thanks for the changes @RenanLukas 🙇 . Approving and merging 👍 !

**Pull request merged**